### DISCUSSION TITLE
Theme Provider

### DISCUSSION THREAD

**Discussion by d0whc3r** [ben-rogerson/twin.macro#272] [Ideas]
Could it be possible to add a "provider" to handle changes of theme elements in runtime?

Something similar to "ThemeProvider" in styled-components, usage example:

```jsx
import tailwindcss from '../tailwindcss/tailwindcss.config';

const darkTheme = {
  ...tailwindcss,
  colors: { ...tailwindcss.colors, primary: 'black' }
};

const lightTheme = {
  ...tailwindcss,
  colors: { ...tailwindcss.colors, primary: 'white' }
};

export default function App() {
return (
  <div>
    <ThemeProvider theme={lightTheme}>
      <span css={css`color: ${theme`colors.primary`};}>sample with white</span>
    </ThemeProvider>
  
    <ThemeProvider theme={darkTheme}>
      <span css={css`color: ${theme`colors.primary`};}>sample with black</span>
    </ThemeProvider>
  </div>
  );
}
```

https://styled-components.com/docs/api#themeprovider

**Comment by d0whc3r** [comment]
Ok... I think this question is answered here: https://github.com/ben-rogerson/twin.macro/discussions/251

**Comment by fvanwijk** [comment]
Yes you can resolve the Tailwind theme and put it into a ThemeProvider, just like the example above.
In tailwind v1 you had the [`resolveConfig` helper](https://v1.tailwindcss.com/docs/configuration#referencing-in-java-script) but it was not the preferred approach and I can't find it in the v2 docs.

How to use Tailwind/twin in combination with `ThemeProvider` is demonstrated in the [advanced](https://github.com/ben-rogerson/twin.macro/blob/master/docs/advanced-theming.md) react+styled-components example.

**Reply by d0whc3r** [reply]
Ok I see, I can change css variables in GlobalStyle to have a dynamic theme
